crates/revm/src/database.rs

```diff
@@ -2,18 +2,19 @@ use reth_interfaces::Error;
 use reth_primitives::{H160, H256, KECCAK_EMPTY, U256};
 use reth_provider::StateProvider;
 use revm::{
-    db::{CacheDB, DatabaseRef},
+    db::DatabaseRef,
     primitives::{AccountInfo, Bytecode},
+    Database, StateDBBox,
 };
 
-/// SubState of database. Uses revm internal cache with binding to reth StateProvider trait.
-pub type SubState<DB> = CacheDB<State<DB>>;
+/// State boxed database with reth Error.
+pub type RethStateDBBox<'a> = StateDBBox<'a, Error>;
 
 /// Wrapper around StateProvider that implements revm database trait
 #[derive(Debug, Clone)]
-pub struct State<DB: StateProvider>(pub DB);
+pub struct RevmDatabase<DB: StateProvider>(pub DB);
```

Comment:
what do you think about

`StateProviderDB`, `StateProviderDatabase` here?

Reply:
Better, will use `StateProviderDatabase`